tests/e2e-instrumentation-test.sh

```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
```

Comment:
Eventually we will have to normalize this file against all e2e tests being added here (or unify these tests with the work being done by @geyslan in the `test-integration` logic).